src/components/policy/policy_external/src/policy_manager_impl.cc

```diff
@@ -399,39 +421,58 @@ void PolicyManagerImpl::ProcessAppPolicyCheckResults(
       continue;
     }
 
+    actions_for_apps[app_id].app_policies = app_policy;
     switch (it_results->second) {
       case RESULT_NO_CHANGES:
         continue;
       case RESULT_APP_REVOKED:
-        NotifySystem(*app_policy);
+        actions_for_apps[app_id].is_notify_system = true;
         continue;
       case RESULT_NICKNAME_MISMATCH:
-        NotifySystem(*app_policy);
+        actions_for_apps[app_id].is_notify_system = true;
         continue;
       case RESULT_CONSENT_NEEDED:
-      case RESULT_PERMISSIONS_REVOKED_AND_CONSENT_NEEDED: {
-        // Post-check after ExternalConsent consent changes
-        const std::string policy_app_id = app_policy->first;
-        if (!IsConsentNeeded(policy_app_id)) {
-          sync_primitives::AutoLock lock(app_permissions_diff_lock_);
-
-          PendingPermissions::iterator app_id_diff =
-              app_permissions_diff_.find(policy_app_id);
-
-          if (app_permissions_diff_.end() != app_id_diff) {
-            app_id_diff->second.appPermissionsConsentNeeded = false;
-          }
-        }
-      } break;
+      case RESULT_PERMISSIONS_REVOKED_AND_CONSENT_NEEDED:
+        actions_for_apps[app_id].is_consent_needed = true;
+        break;
       case RESULT_CONSENT_NOT_REQIURED:
       case RESULT_PERMISSIONS_REVOKED:
       case RESULT_REQUEST_TYPE_CHANGED:
         break;
       default:
         continue;
     }
-    NotifySystem(*app_policy);
-    SendPermissionsToApp(*app_policy);
+    actions_for_apps[app_id].is_notify_system = true;
+    actions_for_apps[app_id].is_send_permissions_to_app = true;
+  }
+
+  // Perform all needed actions once per app_id
+  for (ActionsForApplications::const_iterator it = actions_for_apps.begin();
+       it != actions_for_apps.end();
+       ++it) {
+    policy_table::ApplicationPolicies::const_iterator app_policy =
+        it->second.app_policies;
+
+    if (it->second.is_consent_needed) {
+      // Post-check after ExternalConsent consent changes
+      const std::string policy_app_id = app_policy->first;
```

Comment:
@AKalinich-Luxoft reference

Reply:
@LuxoftAKutsan fixed in https://github.com/smartdevicelink/sdl_core/pull/1779/commits/bfdcdd0f21dc83ffb8a607060d3392cad10406a4